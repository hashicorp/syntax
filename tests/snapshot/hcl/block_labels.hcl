// generic blocks with multiple labels as strings
block0 {}
block0-newline {
}

block1 "foo" {}
block1-newline "foo" {
}

block2 "foo" "bar" {}
block2-newline "foo" "bar" {
}

block3 "foo" "bar" "baz" {}
block3-newline "foo" "bar" "baz" {
}

block4 "foo" "bar" "baz" "lorum" {}
block4-newline "foo" "bar" "baz" "lorum" {
}

block5 "foo" "bar" "baz" "lorum" "ipsum" {}
block5-newline "foo" "bar" "baz" "lorum" "ipsum" {
}

// generic blocks with indentifier labels
blocklabel1 foo bar lorum-ipsum {}
blocklabel1-newline foo bar lorum-ipsum {
}

blocklabelmixed "foo" bar "loru" m-ipsum {}
blocklabelmixed-newline "foo" bar "loru" m-ipsum {
}

// utf8 labels
blockutf8 "foož:/ᚠᚢ" {}
blockutf8-newline "foož:/ᚠᚢ" {
}

// edgecases
blockempty "" {}
blockempty-newline "" {
}

block-single-char-indentifier a {}
block-single-char-indentifier-newline a {
}

byte_match_statement_rules = local.enabled && var.byte_match_statement_rules != null ? {
}

path "secrets/data/users/{{identity.entity.name}}/*" {
  capabilities = ["create", "update", "patch", "read", "delete", "list"]
}

path "secrets/data/users/*" {
  capabilities = ["create", "update", "patch", "read", "delete", "list"]
}