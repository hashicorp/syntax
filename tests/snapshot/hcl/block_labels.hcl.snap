>// generic blocks with multiple labels as strings
#^^ source.hcl comment.line.double-slash.hcl punctuation.definition.comment.hcl
#  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl comment.line.double-slash.hcl
>block0 {}
#^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#      ^ source.hcl meta.block.hcl
#       ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
#        ^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>block0-newline {
#^^^^^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#              ^ source.hcl meta.block.hcl
#               ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>block1 "foo" {}
#^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#      ^ source.hcl meta.block.hcl
#       ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#            ^ source.hcl meta.block.hcl
#             ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
#              ^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>block1-newline "foo" {
#^^^^^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#              ^ source.hcl meta.block.hcl
#               ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                    ^ source.hcl meta.block.hcl
#                     ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>block2 "foo" "bar" {}
#^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#      ^ source.hcl meta.block.hcl
#       ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#            ^ source.hcl meta.block.hcl
#             ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                  ^ source.hcl meta.block.hcl
#                   ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
#                    ^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>block2-newline "foo" "bar" {
#^^^^^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#              ^ source.hcl meta.block.hcl
#               ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                    ^ source.hcl meta.block.hcl
#                     ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                          ^ source.hcl meta.block.hcl
#                           ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>block3 "foo" "bar" "baz" {}
#^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#      ^ source.hcl meta.block.hcl
#       ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#            ^ source.hcl meta.block.hcl
#             ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                  ^ source.hcl meta.block.hcl
#                   ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                        ^ source.hcl meta.block.hcl
#                         ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
#                          ^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>block3-newline "foo" "bar" "baz" {
#^^^^^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#              ^ source.hcl meta.block.hcl
#               ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                    ^ source.hcl meta.block.hcl
#                     ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                          ^ source.hcl meta.block.hcl
#                           ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                ^ source.hcl meta.block.hcl
#                                 ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>block4 "foo" "bar" "baz" "lorum" {}
#^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#      ^ source.hcl meta.block.hcl
#       ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#            ^ source.hcl meta.block.hcl
#             ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                  ^ source.hcl meta.block.hcl
#                   ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                        ^ source.hcl meta.block.hcl
#                         ^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                ^ source.hcl meta.block.hcl
#                                 ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
#                                  ^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>block4-newline "foo" "bar" "baz" "lorum" {
#^^^^^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#              ^ source.hcl meta.block.hcl
#               ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                    ^ source.hcl meta.block.hcl
#                     ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                          ^ source.hcl meta.block.hcl
#                           ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                ^ source.hcl meta.block.hcl
#                                 ^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                        ^ source.hcl meta.block.hcl
#                                         ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>block5 "foo" "bar" "baz" "lorum" "ipsum" {}
#^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#      ^ source.hcl meta.block.hcl
#       ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#            ^ source.hcl meta.block.hcl
#             ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                  ^ source.hcl meta.block.hcl
#                   ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                        ^ source.hcl meta.block.hcl
#                         ^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                ^ source.hcl meta.block.hcl
#                                 ^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                        ^ source.hcl meta.block.hcl
#                                         ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
#                                          ^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>block5-newline "foo" "bar" "baz" "lorum" "ipsum" {
#^^^^^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#              ^ source.hcl meta.block.hcl
#               ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                    ^ source.hcl meta.block.hcl
#                     ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                          ^ source.hcl meta.block.hcl
#                           ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                ^ source.hcl meta.block.hcl
#                                 ^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                        ^ source.hcl meta.block.hcl
#                                         ^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                                ^ source.hcl meta.block.hcl
#                                                 ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>// generic blocks with indentifier labels
#^^ source.hcl comment.line.double-slash.hcl punctuation.definition.comment.hcl
#  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl comment.line.double-slash.hcl
>blocklabel1 foo bar lorum-ipsum {}
#^^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#           ^ source.hcl meta.block.hcl
#            ^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#               ^ source.hcl meta.block.hcl
#                ^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                   ^ source.hcl meta.block.hcl
#                    ^^^^^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                               ^ source.hcl meta.block.hcl
#                                ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
#                                 ^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>blocklabel1-newline foo bar lorum-ipsum {
#^^^^^^^^^^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#                   ^ source.hcl meta.block.hcl
#                    ^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                       ^ source.hcl meta.block.hcl
#                        ^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                           ^ source.hcl meta.block.hcl
#                            ^^^^^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                       ^ source.hcl meta.block.hcl
#                                        ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>blocklabelmixed "foo" bar "loru" m-ipsum {}
#^^^^^^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#               ^ source.hcl meta.block.hcl
#                ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                     ^ source.hcl meta.block.hcl
#                      ^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                         ^ source.hcl meta.block.hcl
#                          ^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                ^ source.hcl meta.block.hcl
#                                 ^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                        ^ source.hcl meta.block.hcl
#                                         ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
#                                          ^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>blocklabelmixed-newline "foo" bar "loru" m-ipsum {
#^^^^^^^^^^^^^^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#                       ^ source.hcl meta.block.hcl
#                        ^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                             ^ source.hcl meta.block.hcl
#                              ^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                 ^ source.hcl meta.block.hcl
#                                  ^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                        ^ source.hcl meta.block.hcl
#                                         ^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                                ^ source.hcl meta.block.hcl
#                                                 ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>// utf8 labels
#^^ source.hcl comment.line.double-slash.hcl punctuation.definition.comment.hcl
#  ^^^^^^^^^^^^ source.hcl comment.line.double-slash.hcl
>blockutf8 "foož:/ᚠᚢ" {}
#^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#         ^ source.hcl meta.block.hcl
#          ^^^^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                    ^ source.hcl meta.block.hcl
#                     ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
#                      ^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>blockutf8-newline "foož:/ᚠᚢ" {
#^^^^^^^^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#                 ^ source.hcl meta.block.hcl
#                  ^^^^^^^^^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                            ^ source.hcl meta.block.hcl
#                             ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>// edgecases
#^^ source.hcl comment.line.double-slash.hcl punctuation.definition.comment.hcl
#  ^^^^^^^^^^ source.hcl comment.line.double-slash.hcl
>blockempty "" {}
#^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#          ^ source.hcl meta.block.hcl
#           ^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#             ^ source.hcl meta.block.hcl
#              ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
#               ^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>blockempty-newline "" {
#^^^^^^^^^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#                  ^ source.hcl meta.block.hcl
#                   ^^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                     ^ source.hcl meta.block.hcl
#                      ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>block-single-char-indentifier a {}
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#                             ^ source.hcl meta.block.hcl
#                              ^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                               ^ source.hcl meta.block.hcl
#                                ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
#                                 ^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>block-single-char-indentifier-newline a {
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl meta.block.hcl entity.name.type.hcl
#                                     ^ source.hcl meta.block.hcl
#                                      ^ source.hcl meta.block.hcl variable.other.enummember.hcl
#                                       ^ source.hcl meta.block.hcl
#                                        ^ source.hcl meta.block.hcl punctuation.section.block.begin.hcl
>}
#^ source.hcl meta.block.hcl punctuation.section.block.end.hcl
>
>byte_match_statement_rules = local.enabled && var.byte_match_statement_rules != null ? {
#^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl variable.declaration.hcl variable.other.readwrite.hcl
#                          ^ source.hcl variable.declaration.hcl
#                           ^ source.hcl variable.declaration.hcl keyword.operator.assignment.hcl
#                            ^ source.hcl variable.declaration.hcl
#                             ^^^^^ source.hcl
#                                  ^ source.hcl keyword.operator.accessor.hcl
#                                   ^^^^^^^ source.hcl variable.other.member.hcl
#                                          ^ source.hcl
#                                           ^^ source.hcl keyword.operator.logical.hcl
#                                             ^^^^ source.hcl
#                                                 ^ source.hcl keyword.operator.accessor.hcl
#                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl variable.other.member.hcl
#                                                                            ^ source.hcl
#                                                                             ^^ source.hcl keyword.operator.hcl
#                                                                               ^ source.hcl
#                                                                                ^^^^ source.hcl constant.language.hcl
#                                                                                    ^ source.hcl
#                                                                                     ^ source.hcl keyword.operator.hcl
#                                                                                      ^ source.hcl
#                                                                                       ^ source.hcl meta.braces.hcl punctuation.section.braces.begin.hcl